eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/Internal/ClientPipelineExtensions.cs

```diff
@@ -5,66 +5,73 @@
 
 #nullable disable
 
-using System.ClientModel;
-using System.ClientModel.Primitives;
 using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
 
 namespace BasicTypeSpec
 {
     internal static partial class ClientPipelineExtensions
     {
-        public static async ValueTask<PipelineResponse> ProcessMessageAsync(this ClientPipeline pipeline, PipelineMessage message, RequestOptions options)
+        public static async ValueTask<Response> ProcessMessageAsync(this HttpPipeline pipeline, HttpMessage message, RequestContext options)
         {
-            await pipeline.SendAsync(message).ConfigureAwait(false);
+            await pipeline.SendAsync(message, default).ConfigureAwait(false);
 
-            if (message.Response.IsError && (options?.ErrorOptions & ClientErrorBehaviors.NoThrow) != ClientErrorBehaviors.NoThrow)
+            if (message.Response.IsError && (options?.ErrorOptions & ErrorOptions.NoThrow) != ErrorOptions.NoThrow)
             {
-                throw await ClientResultException.CreateAsync(message.Response).ConfigureAwait(false);
+                throw new RequestFailedException(message.Response);
             }
 
-            PipelineResponse response = message.BufferResponse ? message.Response : message.ExtractResponse();
+            Response response = message.BufferResponse ? message.Response : ExtractResponseContent(message);
```

Comment:
We should sync on ExtractResponseContent ... this may not be the right pattern for Azure clients.